src/controller/p2p.rs

```diff
@@ -121,6 +151,41 @@ impl<EventStream: Stream<Item = WorkerEvent>> P2PController<EventStream> {
         info!("Heartbeat processing task finished");
     }
 
+    async fn run_assignments_loop(&self, cancellation_token: CancellationToken, assignment_check_interval: Duration) {
+        let mut timer =
+            tokio::time::interval_at(tokio::time::Instant::now(), assignment_check_interval);
+        timer.set_missed_tick_behavior(MissedTickBehavior::Delay);
+        IntervalStream::new(timer)
+            .take_until(cancellation_token.cancelled_owned())
+            .for_each(|_| async move {
+                tracing::debug!("Checking assignment");
+                let network_state_filename = match self.network {
+                    Network::Tethys => "network-state-tethys.json",
+                    Network::Mainnet => "network-state-mainnet.json",
+                };
+                let network_state_url = format!("https://metadata.sqd-datasets.io/{network_state_filename}");
+                let mut latest_assignment = self.latest_assignment.lock().unwrap();
+                if let Ok(assignment_option) = Assignment::from_url(network_state_url, latest_assignment.clone()).await {
+                    if let Some(assignment) = assignment_option {
+                        let peer_id = self.worker.peer_id.unwrap();
+                        let private_key = self.private_key.clone();
+                        let calculated_chunks = assignment.dataset_chunks_for_peer_id(peer_id.to_string()).unwrap();
+                        let headers = assignment.headers_for_peer_id(peer_id.to_string(), private_key).unwrap();
```

Comment:
You should only call `unwrap` if the logic that guarantees that its safety is located in the same function. But here the result depends on the file from the internet, so it should be handled by returning an `Err` and logging it instead of panicking.

Reply:
done